types/node-rsa/index.d.ts

```diff
@@ -0,0 +1,197 @@
+// Type definitions for node-rsa 0.4
+// Project: https://github.com/rzcoder/node-rsa
+// Definitions by: Ali Taheri <https://github.com/alitaheri>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/// <reference types="node" />
+
+declare class NodeRSA {
+    /**
+     * Generate new key with length specified.
+     */
+    constructor(key?: NodeRSA.KeyBits);
```

Comment:
But it is. they are both optional.

If I merge them, then there is no way to be explicit about the difference in both documentation and argument count.

If the developer writes `new NodeRSA({b: 512})` and they get suggestion about the other two arguments they will get confused.

Although, they should read the documentation anyway. Since `KeyComponents` does not include a property with the name `b`, there is no way the second overload gets chosen and the dev sees the correct info.

I'd say if such overloads won't cause problems ( since there is no intersection between the types ), let them be. But if they do (or if it's inconvenient), I'll happily merge the two. Your call 😁